### PR TITLE
HPCC-17569 Add roxie support for persistent http connections

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by richardkchapman** [comment]
These smoketest errors look unrelated. @AttilaVamos can you check?

File: common/thorhelper/roxiehelper.cpp [hunk 1]
```diff
@@ -1598,15 +1598,21 @@ void CSafeSocket::setHttpMode(const char *queryName, bool arrayMode, HttpHelper
...
+
     if (mlResponseFmt==MarkupFmt_JSON)
     {
         contentHead.set("{");
         contentTail.set("}");
     }
     else
     {
-        StringAttrBuilder headText(contentHead), tailText(contentTail);
+        StringAttrBuilder headText(contentHead.clear()), tailText(contentTail.clear());
```
Comment (AttilaVamos) [inline common/thorhelper/roxiehelper.cpp]: If I see well, the contentHead.clear() has no return value (void) and a StringAttrBuilder object needs a reference. 
I think the proper way is:
contentHead.clear();
contentTail.clear();
StringAttrBuilder headText(contentHead), tailText(contentTail);
(And it is build, I tested.)

**Review by AttilaVamos — CHANGES_REQUESTED** [review]
There is one problem which prevents to build your code.

**Comment by afishbeck** [comment]
@richardkchapman Updated.

**Pull request closed without merging**